src/NSubstitute.Analyzers.Shared/Extensions/IArgumentOperationExtensions.cs

```diff
@@ -0,0 +1,26 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace NSubstitute.Analyzers.Shared.Extensions
+{
+    internal static class IArgumentOperationExtensions
+    {
+        public static ITypeSymbol GetArgumentOperationActualTypeSymbol(this IArgumentOperation argumentOperation)
```

Comment:
This can be simplified:

```csharp
public static ITypeSymbol GetArgumentOperationActualTypeSymbol(this IArgumentOperation argumentOperation) =>
    argumentOperation.Value switch {
        IConversionOperation conversionOperation => conversionOperation.Operand.Type,
        _ => argumentOperation.GetArgumentOperationDeclaredTypeSymbol()
    };
```

Or using older syntax:

```csharp
public static ITypeSymbol GetArgumentOperationActualTypeSymbol(this IArgumentOperation argumentOperation) {
    switch (argumentOperation.Value) {
        case IConversionOperation conversionOperation: return conversionOperation.Operand.Type;
        default: return argumentOperation.GetArgumentOperationDeclaredTypeSymbol();
    }
}
```

I think this is clearer than updating the local `conversionTypeSymbol` variable?

Reply:
This is not quite the same. In my implementation when `conversionOperation.Operand.Type` is null I have a fallback to `argumentOperation.GetArgumentOperationDeclaredTypeSymbol()`. I dont know for 100% if `conversionOperation.Operand.Type` can be null, but because of the fact that `conversionOperation.Operand` is `IOperation` which in documentation says
https://github.com/dotnet/roslyn/blob/38718c45ef8add102834a42d96830901b68515bb/src/Compilers/Core/Portable/Operations/IOperation.cs#L35
>         /// Result type of the operation, or null if the operation does not produce a result.
I think it is safer to leave the fallback as it is

Reply:
Ah I see, thanks for pointing that out 👍